core/domain/auth_jobs_one_off.py

```diff
@@ -0,0 +1,195 @@
+                ', '.join(sorted(user_ids)))
+
+
+class PopulateFirebaseAccountsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job that maps over UserSettingsModels and imports them to Oppia's
+    Firebase server. The latter requires that we "specify [an ID] at a minimum",
+    so we use Oppia's user_id as the value.
+
+    NOTE: **DO NOT** ASSUME THAT FIREBASE IDS AND OPPIA USER IDS WILL BE THE
+    SAME! We are only doing this for users that already exist; future users that
+    sign up with Firebase will have an entirely different ID.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user):
+        auth_id = firebase_auth_services.get_auth_id_from_user_id(user.id)
+        if auth_id is not None:
+            yield (POPULATED_KEY, 1)
+        else:
+            yield (
+                NOT_POPULATED_KEY,
+                (_strip_uid_prefix(user.id), user.id, user.email, user.deleted))
+
+    @staticmethod
+    def reduce(key, values):
+        if key == POPULATED_KEY:
+            yield (AUDIT_KEY, len(values))
+            return
+
+        try:
+            app = firebase_admin.initialize_app()
+        except Exception as exception:
+            yield (WARNING_KEY, repr(exception))
+            return
+
+        # NOTE: This is only sorted to make unit testing easier.
+        user_fields = sorted(ast.literal_eval(v) for v in values)
+        user_records = [
+            firebase_auth.ImportUserRecord(
+                uid=auth_id, email=email, disabled=disabled,
+                email_verified=True)
+            for auth_id, _, email, disabled in user_fields
+        ]
+
+        # The Firebase Admin SDK places a hard-limit on the number of users that
+        # can be "imported" in a single call. To compensate, we break up the
+        # users into chunks.
+        offsets = python_utils.RANGE(
+            0, len(user_records), FIREBASE_IMPORT_USERS_MAX_LEN)
+        results = (
+            _populate_firebase([record for record in record_group if record])
+            for record_group in _grouper(
+                user_records, FIREBASE_IMPORT_USERS_MAX_LEN))
+
+        assocs_to_create = []
+        for offset, (result, exception) in python_utils.ZIP(offsets, results):
+            if exception is not None:
+                yield (FAILURE_KEY, repr(exception))
+            else:
+                successful_indices = set(python_utils.RANGE(
+                    result.success_count + result.failure_count))
+                for error in result.errors:
+                    successful_indices.remove(error.index)
+                    debug_info = (
+                        'Import user_id=%r failed: %s' % (
+                            user_fields[offset + error.index][1], error.reason))
+                    yield (FAILURE_KEY, debug_info)
+                assocs_to_create.extend(
```

Comment:
Should we also be doing this with deleted models?

Reply:
@vojtechjelinek what do you think? My understanding is that deleted models are how wipeout keeps track of what has been processed.

Reply:
Hmm, lets use `AuditUserEmailsOneOffJob` to make sure that there are no `UserSettingsModel`s with `deleted` set to `True` (this means that the wipeout is in progress for that user), if the user is fully deleted there are no models for him. And here just to make sure do not create anything for users marked as deleted, just skip them.

Reply:
Done.